expression/util.go

```diff
@@ -511,7 +511,8 @@ func DatumToConstant(d types.Datum, tp byte) *Constant {
 }
 
 // GetParamExpression generate a getparam function expression.
-func GetParamExpression(ctx sessionctx.Context, v *driver.ParamMarkerExpr, useCache bool) (Expression, error) {
+func GetParamExpression(ctx sessionctx.Context, v *driver.ParamMarkerExpr) (Expression, error) {
+	useCache := ctx.GetSessionVars().StmtCtx.UseCache
```

Comment:
useCache is always false since in `cacheable_checker.go` we set cacheable to false when there's ParamMarker in order by, group by and limit?

Reply:
Yes. Since ``expression.GetParamExpression()`` is used commonly, ``useCache`` must be checked for each statement:
https://github.com/pingcap/tidb/blob/497fa925756716a6937d767713cb86f3b6ef90fe/planner/core/expression_rewriter.go#L757-L763

Reply:
Ok, got it.